### ISSUE TITLE
Controls inside of alerts don't make sense with screen readers announcements

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3549]
## Describe the bug

When controls are inside live regions, their accessible names are announced but they don't make sense when combined with the content and there is no way for the screen reader users to immediately activate these controls.

## How to reproduce

https://stackblitz.com/github/clane/clarityTest2/tree/master/clarity-test-app

- Using a screen reader, activate the 'Alerts- Standard Alert' link under the heading 'Alert Demos'
- Observe that the screen reader announces the name of the 'Shutdown' link and that it does not make sense when combined with the alert text
- Observe that you do not hear any instructions on how to activate the 'Shutdown' functionality

## Expected behavior

TBD

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

Possible solutions:

The simplest solution would be to implement the alerts that contain controls like a modal dialog rather than a live region. This will force screen reader users to find the newly added controls and read the newly added content. 

Otherwise, if a live region must be used to announce the presence of controls, we should implement keyboard shortcuts to provide the same functionality offered by the newly added controls and announce those keyboard shortcuts to screen reader users. These keyboard shortcuts will also have to be documented. Since the accessible names of the controls are announced as well, accessible hidden text can be added if necessary to make the content and control names make sense when combined.

**Comment by mathisscott** [comment]
1. Controls can be dropdowns, buttons, or links. We support no other controls inside of alerts. We should be using buttons for actions, links for navigation.
2. The text for these controls should be more descriptive. This is an application issue but we could/should provide better examples.

**Comment by mathisscott** [comment]
We need to update the documentation to recommend _against_ using controls inside of alerts. Alerts/notifications that require controls inside of them need a modal-based flow of focus. This is not feasible inside of alerts.

The recommendation/guideline in our documentation should point out that controls inside of alerts are not accessible. When toasts/snackbars are added, that is when we will manage flow of control (preferably for screen-readers only) to allow for screen-reader users to jump to notifications and interact with the controls.

This is not something we can fix by Aug 31.

**Comment by Jinnie** [comment]
cherrypicked to v2 with f2216475226219605d8e8feae779c31d3f5e523e
done
